### ISSUE TITLE
Abstract image loading

### ISSUE THREAD

**Issue by zsoltk** [VEINHORN/ScrollGalleryView#12]
Hello there, great lib! 

What first came to my mind is that loading the image could be abstracted away through an interface, so that client code can use whatever technique it wants (popular libraries like Picasso, Glide, Universal Image Loader, Fresco, etc.) instead of directly loading a Bitmap.

So rather than using ```addImage(Bitmap image)```, it would be ```addImage(ScrollGalleryViewItem item)```, where:

```
interface ScrollGalleryViewItem {
    View getView();
    View getThumbnail(int thumbnailSize);
}
```
(Just a rough sketch, but you get the idea)

Also, there would be a default implementation of this interface ```SimpleScrollGalleryViewItem```, which simply does what the lib currently does, so that people can easily use the lib without any extra code right away.

What do you think?

Ps.: :+1: for the nice, clean code :)

**Comment by VEINHORN** [comment]
@zsoltk Thank you. :+1:

**Comment by inver** [comment]
@zsoltk I improved library for loading media through a loader. You can check it in my fork.

**Comment by zsoltk** [comment]
@inver did this eventually get merged in #21 ?

**Comment by inver** [comment]
@zsoltk Yes
